src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String MESSAGE_OUT_OF_BOUNDS
+            = "The index passed to any of the methods is invalid";
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(T value) {
+            this.value = value;
+        }
```

Comment:
The `Node` constructor should accept all fields (`prev`, `value`, `next`) as parameters, as suggested by the checklist.